Sources/MongoSwift/MongoCollection+Write.swift

```diff
@@ -191,8 +219,33 @@ extension MongoCollection {
             // TODO SWIFT-139: include writeErrors and writeConcernErrors from reply in the error
             throw MongoError.commandError(message: toErrorString(error))
         }
+
+        guard isAcknowledged(options?.writeConcern) else {
+            return nil
+        }
+
         return try BsonDecoder().decode(DeleteResult.self, from: reply)
     }
+
+    /**
+     * Returns whether the operation's write concern is acknowledged. If `nil`,
+     * the collection's write concern will be considered.
+     *
+     * - Parameters:
+     *   - writeConcern: `WriteConcern` from the operation's options struct
+     *
+     * - Returns: Whether the operation will use an acknowledged write concern
+     */
+    fileprivate func isAcknowledged(_ writeConcern: WriteConcern?) -> Bool {
+        /* If the collection's write concern is also `nil` it is the default. We
+         * can safely assume it is acknowledged, sincethe server requires that
+         * getLastErrorDefaults is acknowledged by at least one member. */
```

Comment:
FYI: I tested this locally using MongoDB 4.0 where it would not allow my to reconfigure the replica set with an unacknowledged write concern (see: [`settings.getLastErrorDefaults`](https://docs.mongodb.com/manual/reference/replica-configuration/#rsconf.settings.getLastErrorDefaults)). Therefore, I think we can safely assume that all default write concerns are acknowledged.

I didn't test earlier versions, but I doubt that restriction is new to 4.0.

Reply:
sounds like a reasonable assumption to me.